File: codes/go/service/postgres.go
package service

import (
	"context"

	"github.com/go-pg/pg/v10"

	"github.com/nicolas-grevin/green-it/util"
)

type PostgresService struct {
  DB *pg.DB
}

func NewPostgresService() *PostgresService {
  db := pg.Connect(&pg.Options{
    Addr:       util.GetEnvDefault("POSTGRES_HOST", "localhost") + ":" + util.GetEnvDefault("POSTGRES_POST", "5432"),
    User:       util.GetEnvDefault("POSTGRES_USER", "user"),
    Password:   util.GetEnvDefault("POSTGRES_PASSWORD", "password"),
    Database:   util.GetEnvDefault("POSTGRES_DB", "database"),
  })

  return &PostgresService{DB: db}
}

func (ps *PostgresService) Ping() error {
  return ps.DB.Ping(context.Background())
}

File: codes/deno/service/postgres.ts
import { Client } from "postgres";

export class PostgresService {
  private _client: Client;

  constructor() {
    this._client = new Client({
      hostname: Deno.env.get("POSTGRES_HOST") || "localhost",
      port: Deno.env.get("POSTGRES_PORT") || 5432,
      database: Deno.env.get("POSTGRES_DB") || "database",
      user: Deno.env.get("POSTGRES_USER") || "user",
      password: Deno.env.get("POSTGRES_PASSWORD") || "password",
    });
  }

  getClient(): Client {
    return this._client;
  }
}
